qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/plot_presenter.py

```diff
@@ -102,6 +103,9 @@ def do_serial_fit(self):
     def do_seq_fit(self):
         self.fit_all_started_notifier.notify_subscribers(True)
 
+    def toggle_legend(self):
+        self.view.update_figure()
```

Comment:
Just wondering, is there a reason we need to update the whole figure on the legend visibility being changed? Could we call `view.update_legend()` (which atm requires an axes object be passed, but we could get this inside the function itself)?

Reply:
I did try to do just `view.update_legend()`, but it didn't work the first try. I suspect because I didn't include `view.figure.canvas.draw()`, so it didn't draw the plot again. Maybe @warunawickramasingha can try this.

Reply:
I have updated the PR inline with the comments. @RichardWaiteSTFC I have moved the slot implementation to the view itself since the presenter had not much to do except only calling back methods in the view class. This results in almost all of the changes to be in the view class and hence the unit test for the presenter had to be removed. 

@GuiMacielPereira  thanks for noting that view.figure.canvas.draw() should be called to update the legend after toggling the legend visibility.

Reply:
All looks good. The connection between toggling the legend and updating the legend is still done in the presenter (this is in line with how other connections are set up), but the function call is now inside the view. I tried the toggle button manually and it is working.